bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/categorylist/CategoriesRetriever.java

```diff
@@ -32,6 +39,10 @@ protected CategoryTreeQueryDefinition generateCategoryQuery() {
                 .position()
                 .image();
 
+            if (enableUIDSupport || identifierType == UrlProvider.CategoryIdentifierType.UID) {
```

Comment:
We should also only query `id` if we are using it. See line 36.

Reply:
The condition here is for if it's supported to query a category by UID. So you can query for the UID anytime, no condition needed.